### PR TITLE
Virology Rework 2023 - Threshold Randomization

### PR DESCRIPTION
**Pull request by Job-IDev**
## About The Pull Request

This PR changes the way virology is played. Currently, virology is a lot of spreadsheet gameplay.
The average round of a virologist looks like this:

1. Collect symptoms (either in maintenance, from people or with chemistry)
2. Go through your collection of symptoms
3. Look at your spreadsheet
4. Create your default healing/enhancement virus

While this is enjoyable for the first couple of rounds, it's mostly a static role.
This is where this PR comes to play.

By randomizing the symptom thresholds you cannot simply go off your list and create the same virus over and over again. The randomization will generate within a predefined range. (Balanced according to their severity/deadliness) Any medical personnel is able to download the newly added "Virology Symptoms" app on their PDA. Within the app, you can search for known symptoms and the thresholds they have for this round.

## Why It's Good For The Game

This PR will add some diversity to virology. Instead of mostly spreadsheet gaming it will require someone to look at the pandemic or PDA app if they want to know the thresholds. (Alternatively, you can just use symptoms that meet the maximum threshold requirement)

It will open new possibilities for new mixes. Mixes that were previously impossible to create will now have a random chance to become possible. Or if you're unlucky mixes that were previously possible are now impossible to make. These thresholds are only bound to one round and will be refreshed every new round.

## Testing Photographs and Procedure

<details>
<summary>Screenshots</summary>

Only Medbay Access

Downloaded App

App overview

App Search Function

</details>

## Changelog
:cl:
add: Added randomization for virus thresholds
add: New PDA app for virus symptoms
tweak: Tweaked the pandemic to display the randomized thresholds correctly
spellcheck: Fixed some typos in threshold descriptions & virus descriptions
/:cl:

### CODE REVIEW

**Comment by Job-IDev** [comment]
Please hold while I try to fix this tgui issue...

**Review by PowerfulBacon — COMMENTED** [review]
Use defines instead of magic strings for the thresholds
```
#define THRESHOLD_STEALTH
get_threshold(THRESHOLD_STEALTH)
```

**Comment by PowerfulBacon** [comment]
Viro is still going to suck as a role, but this is a good change that makes it so spreadsheet gaming is encouraged less and might allow for some more interesting.
It might also be useful to have an ability to sort the symptoms by their stats on the PDA, so you can find which one has the best infectivity?

**Comment by Job-IDev** [comment]
> It might also be useful to have an ability to sort the symptoms by their stats on the PDA, so you can find which one has the best infectivity?

Tgui is weird. But there might be a way to sort on specific stats.
(I first have to overcome this error headache)

**Comment by Job-IDev** [comment]
Finaly. It's TGUI compliant. 🙂

**Comment by CydiaLamiales** [comment]
I certainly hope you'll update the wiki accordingly.

**Comment by zeskorion** [comment]
this is a terrible idea. it'll absolutely kill viro, as recipes take a long time to make, and you cant make them in game easily due to this

**Comment by zeskorion** [comment]
the best way to implement this are bonus datums on top of symptoms that add new effects(also meaning that symptoms no one uses will suddenly sometimes have optimal use-cases) instead of just randomizing existing symptoms. making viro MORE of a terrible rngfest would kill what few players enjoy it as it is, and make it far less friendly for new players

**Comment by zeskorion** [comment]
I'll write up a cohesive essay on why this is bad after I get my day started and turn on my computer.
 I will note the fact that this allows for much lower thresholds,  and doesn't change the basic spreadsheet sort of gameplay, leads me to believe this is made because the creator has a few recipes he can't quite get tow work. 
If you want to remove spreadsheet viro, /tg/style viro doesn't work at all- port /vg/ style pathology.  this is a bandage fix that feels like a buff in disguise

**Comment by zeskorion** [comment]
Alright, now that i'm at my computer:
This change is not going to accomplish its intent of trying to kill spreadsheet viro. /tg/ style viro is inherently spreadsheet/ predesigned recipe based. Designing viruses is a time consuming process, and this is more likely to just result in people having more spreadsheets with slight variation, or spreadsheets with overcompensation as far as stats go. Non-spreadsheet viro will still *suck*

I prefer spreadsheet style viro, yes, whatever. But if you're going to remove that, or at least try to, instead of shooting for a middle ground which will just make the playstyle viro is designed around less satisfying.
this won't make non-spreadsheet viro more bearable- you'll just have slightly different numbers, and a still tedious disease to work with


I have two alternate propositions to this one, that arent such bandage solutions

the first is thus:
random symptom mutations. When a roundstart virus is generated, it has a chance to put a mutation on one of its symptoms. These mutations are stuck to that particular copy of the symptom. They may add a new effect, threshold, or change the virus stats, though they always have an effect that can be seen as useful. This will add a modifier to the symptom's name

this solution is a middle-ground. It'll result in spreadsheet viro still being just as viable (unless the stats of a copy of an important lv9 symptom changes, but whatever, that's fine), but will reward people who deviate from spreadsheets with interesting effects that cannot otherwise be obtained

The second is:
just totally rework viro.  I made beeviro on top of /tg/viro, and it still has many of the intrinsic gameplay flaws. if you wanna remove spreadsheet gameplay, the entire concept of symptoms with stats needs to be removed and reworked. I do not think viro has good gameflow or cohesive design. I did not make it with good knowledge of game design. A total ground-up rework for the department would probably do it good, and make it a lot better for new and old players alike. it'll piss off spreadsheet users, myself included, but a lot less than if viro is just made overall less satisfying for everyone

**Comment by zeskorion** [comment]
tl;dr
either totally rework viro, or make changes that try to make non-spreadsheet viro easier, without impacting spreadsheet gameplay. This isn't necessarily going to make non-spreadsheet viro easier, it's just gonna make the current viable strategy viro is designed for harder

**Review by itsmeow — CHANGES_REQUESTED** [review]
The PDA app's design is literally boxes of text and it hurts my soul to look at.
At minimum add some bolding, preferable use sections, titles, boxes etc to make a sensible UI layout.

I also don't think this is really addressing the issue of viro being spreadsheety since the problem with viro is more it fundamentally does not work, the only part of viro that is good right now is developing cures for random viruses. Making viruses has very little intrinsic pluses and is very monotonous.

**Comment by Job-IDev** [comment]
> The PDA app's design is literally boxes of text and it hurts my soul to look at. At minimum add some bolding, preferable use sections, titles, boxes etc to make a sensible UI layout.

TGUI is pain. I have never worked with it before and I used the records app as a reference.
I'd first like to get an approval on main part of this PR before I start improving the PDA app.

**Comment by EvilDragonfiend** [comment]
In my opinion, randomisation should be applied to stats, not to thresholds.

**Comment by zeskorion** [comment]
> In my opinion, randomisation should be applied to stats, not to thresholds. fixated stats will still keep spreadsheet play.
see what I wrote. making spreadsheeting harder wont make viro any better- it's balanced around it being the only really viable, intended approach. you'd either need to make non-spreadsheeting viable, or totally rework things

**Comment by PowerfulBacon** [comment]
A solution which makes *some* minor improvements over a system with a poor foundation is fine, its unfair to tell them just to rework all of viro.
I do like the idea of finding some symptoms that are really good in one round and using them. It might not kill spread-sheet viro but if it messes up some super killing murderbone virus in a single round, then its a win

**Comment by Job-IDev** [comment]
Alright, so. After a good night of sleep and reading your replies to this PR, I've noticed one similarity on which we can all agree. Virology right now is flawed and will need some changes. Unfortunately, nobody is brave enough to take on this challenge.

Although I only have roughly two weeks of coding experience within DreamMaker Script, I did want to take on this challenge.
My main goal is to make virology more enjoyable for new players and more experienced players. (like myself)

While the current strain of virology does work, it heavily relies on creating a spreadsheet with a combination of the best symptoms. Instead, I think it should rely on what symptoms work best during the round you're playing. This will make virologists have to think on their feet to achieve their goals.

During my last PR, I clearly stepped on some toes, while this PR was received very negatively it did stir up a good discussion about virology in its current state. In this discussion, people mentioned that virology could use changes to symptoms itself.

I quite liked this idea and began working on this project. I figured tweaking the thresholds would be the best approach.

Even though these changes might not get rid of spreadsheet gaming as a whole, I think it will greatly reduce the usage. Additionally, I've created a nice way to get an overview of all available symptoms and their respective thresholds.

Overall, the introduction of random thresholds for symptoms will be a positive change for this server. It adds an element of unpredictability and challenge to the game.

**Comment by zeskorion** [comment]
What I'm saying here is that this is not only not going to kill spreadsheet viro, it's **not going to make viro any easier without spreadsheets**. viro sucks without spreadsheets. 

the goal here needs to be **making viro easier without spreadsheets**.  not making spreadsheets harder as a whole- the latter won't cause the former



having two weeks of experience is a bit sparse for a whole department, no?

**Comment by Job-IDev** [comment]
> having two weeks of experience is a bit sparse for a whole department, no?

That's not what I said at all... I said I only have roughly two weeks of experience with **coding** in DreamMaker.
I have over 200 hours in viro alone...

**Comment by zeskorion** [comment]
> > having two weeks of experience is a bit sparse for a whole department, no?
> 
> That's not what I said at all... I said I only have roughly two weeks of experience with **coding** in DreamMaker. I have over 200 hours in viro alone...

I'm aware of what you mean. reworking an entire department takes a lotta effort tho- more than just a change like this
